sympy/stats/crv.py

```diff
@@ -531,11 +531,14 @@ def compute_density(self, expr, **kwargs):
 
         gs = solveset(expr - y, self.value, S.Reals)
 
-        if isinstance(gs, Intersection) and S.Reals in gs.args:
-            gs = list(gs.args[1])
-
-        if not gs:
-            raise ValueError("Can not solve %s for %s"%(expr, self.value))
+        if isinstance(gs, Intersection):
+            if len(gs.args) == 2:
+                if gs.args[0] is S.Reals:
+                    gs = gs.args[1]
+                else:
+                    raise NotImplementedError("Anticipating Intersection(Reals, Set)")
+        if not gs.is_FiniteSet:
+            raise ValueError("Can not solve %s for %s" % (expr, self.value))
```

Comment:
I still don't see the point of raising NotImplementedError. The line below will raise ValueError if `gs` is an intersection. Nothing else needs to be done.

Reply:
Thanks, I have removed it.